src/components/ThumbnailsPanel/ThumbnailsPanel.js

```diff
@@ -393,11 +428,40 @@ class ThumbnailsPanel extends React.PureComponent {
     );
   }
 
+  toggleDocumentControl = shouldShowControls => {
+    this.props.setSelectedPageThumbnails([]);
+    this.setState({
+      isDocumentControlHidden: !shouldShowControls,
+    });
+  }
+
   render() {
-    const { isDisabled, totalPages, display, isThumbnailControlDisabled } = this.props;
-    const { numberOfColumns, height, width } = this.state;
+    const { isDisabled, totalPages, display, isThumbnailControlDisabled, selectedPageIndexes } = this.props;
+    const { numberOfColumns, height, width, documentControlHeight, isDocumentControlHidden } = this.state;
     const thumbnailHeight = isThumbnailControlDisabled ? 200 : 230;
 
+    const shouldShowControls = !isDocumentControlHidden || selectedPageIndexes.length > 0;
+
+    const documentControl =
```

Comment:
I moved it to inline so you can see what it look like. I think inline is nicer then a `renderDocumentControl` function. I had it as a variable before because I was moving DOM elements around before and it was easier to do it when the `documentControl` was a variable

Reply:
I think this looks ok. I noticed that the indentation is 16 instead of 14 for the <Measure>...</Measure> but I guess it's not a big deal.